controllers/managedocs_controller.go

```diff
@@ -730,6 +758,15 @@ func (r *ManagedOCSReconciler) reconcileAlertmanagerConfig() error {
 				receiver.PagerDutyConfigs[0].Details[0].Value = r.SOPEndpoint
 			case "DeadMansSnitch":
 				receiver.WebhookConfigs[0].URL = &dmsURL
+			case "SendGrid":
+				if notificationEmailValue != "" {
+					receiver.EmailConfigs[0].Smarthost = fmt.Sprintf("%s:%s", smtpSecretData["host"], smtpSecretData["port"])
+					receiver.EmailConfigs[0].AuthUsername = string(smtpSecretData["username"])
+					receiver.EmailConfigs[0].AuthPassword.LocalObjectReference.Name = r.SMTPSecretName
+					receiver.EmailConfigs[0].AuthPassword.Key = "password"
+					receiver.EmailConfigs[0].From = r.AlertSMTPFrom
+				}
```

Comment:
The above code assumes that you can opt in but not opt-out. 
I think it will be best to always configure the secret and only empty/fill the email list to accomplish both opt-in and opt-out functionality.